tapgarden/custodian.go

```diff
@@ -580,12 +585,20 @@ func (c *Custodian) mapProofToEvent(p proof.Blob) error {
 
 	// Check if any of our in-flight events match the last proof's state.
 	for _, event := range c.events {
-		if AddrMatchesAsset(event.Addr, &lastProof.Asset) {
+		if AddrMatchesAsset(event.Addr, &lastProof.Asset) &&
+			event.Outpoint == lastProof.OutPoint() {
+
 			// Importing a proof already creates the asset in the
 			// database. Therefore, all we need to do is update the
 			// state of the address event to mark it as completed
 			// successfully.
-			return c.setReceiveCompleted(event, lastProof, file)
+			err = c.setReceiveCompleted(event, lastProof, file)
+			if err != nil {
+				return fmt.Errorf("error updating event: %w",
+					err)
+			}
+
+			delete(c.events, event.Outpoint)
```

Comment:
We used to `return` here after finding a single matching event. I think it's better as you have it now.